uniffi_bindgen/src/bindings/swift/templates/RustBufferHelper.swift

```diff
@@ -120,196 +120,199 @@ class Writer {
     }
 }
 
-// Types conforming to `Liftable` know how to deserialize ("lift") themselves
-// from a byte buffer. This is equivalent to the `Liftable` trait on the Rust
-// side.
-protocol Liftable {
-    static func lift(from: Reader) throws -> Self
-}
 
-// Types conforming to `Lowerable` know how to serialize ("lower") themselves
-// into a byte buffer. Equivalent to the `Lowerable` trait on the Rust side.
-protocol Lowerable {
-    func lower(into: Writer)
+// Types conforming to `Serializable` can be read and written in a bytebuffer.
+protocol Serializable {
+    func write(into: Writer)
+    static func read(from: Reader) throws -> Self
 }
 
-extension String: Liftable, Lowerable {
-    static func fromFFIValue(_ v: UnsafeMutablePointer<CChar>) throws -> Self {
-        defer {
-            {{ ci.ffi_string_free().name() }}(v)
-        }
-        return String(cString: v)
-    }
-    func toFFIValue() -> UnsafeMutablePointer<CChar> {
-        var rustErr = NativeRustError(code: 0, message: nil)
-        let rustStr = {{ ci.ffi_string_alloc_from().name() }}(self, &rustErr)
-        if rustErr.code != 0 {
-            fatalError("caught a panic while passing a string across the ffi")
-        }
-        return rustStr
-    }
-
-    static func lift(from buf: Reader) throws -> Self {
-        let len: UInt32 = try buf.readInt()
-        return String(bytes: try buf.readBytes(count: Int(len)), encoding: String.Encoding.utf8)!
-    }
-
-    func lower(into buf: Writer) {
-        let len = UInt32(self.utf8.count)
-        buf.writeInt(len)
-        buf.writeBytes(self.utf8)
-    }
+// Types confirming to `ViaFfi` can be transferred back-and-for over the FFI.
+// This is analogous to the Rust trait of the same name.
+protocol ViaFfi: Serializable {
```

Comment:
I've tried to make this map more directly to the matching traits on the rust side, @linacambridge could you please check that I haven't totally destroyed your beautiful creation here?

Reply:
This is exactly what I had in mind with #29, thanks! 💯